django/db/models/fields/related.py

```diff
@@ -731,26 +732,23 @@ def get_reverse_path_info(self):
         pathinfos = [PathInfo(from_opts, opts, (opts.pk,), self.remote_field, not self.unique, False)]
         return pathinfos
 
-    def get_lookup(self, lookup_name):
-        if lookup_name == 'in':
-            return RelatedIn
-        elif lookup_name == 'exact':
-            return RelatedExact
-        elif lookup_name == 'gt':
-            return RelatedGreaterThan
-        elif lookup_name == 'gte':
-            return RelatedGreaterThanOrEqual
-        elif lookup_name == 'lt':
-            return RelatedLessThan
-        elif lookup_name == 'lte':
-            return RelatedLessThanOrEqual
-        elif lookup_name == 'isnull':
-            return RelatedIsNull
-        else:
-            raise TypeError('Related Field got invalid lookup: %s' % lookup_name)
-
-    def get_transform(self, *args, **kwargs):
-        raise NotImplementedError('Relational fields do not support transforms.')
+    def _get_lookup(self, lookup_name):
+        try:
+            return self.class_lookups[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class' class_lookups.
+            for parent in inspect.getmro(self.__class__):
+                # Related lookups need to be explicitly registered to related fields.
+                if parent is ForeignObject:
+                    break
+                if 'class_lookups' not in parent.__dict__:
```

Comment:
Two ideas:
- Add `bases` argument to `_get_lookup()`. This might be a good short term solution, since it only affects an internal method. 
  ```py
  # RegisterLookupMixin
      def _get_lookup(self, lookup_name, bases=None):
          ...
          if bases is None:
              bases = inspect.getmro(self.__class__)
          for parent in bases:
              ...

  # ForeignObject
      def _get_lookup(self, lookup_name, bases=None):
          if bases is None:
              bases = inspect.getmro(self.__class__)
              bases = bases[:bases.index(ForeignObject)+1]
          return super(ForeignObject, self)._get_lookup(lookup_name, bases)
  ```
- Add `RegisterLookupMixin.get_lookups()`. I kind of like this approach because it makes introspecting available lookups & transforms easier. It might also make `_get_lookup()` unnecessary.
  ```py
  def merge_dicts(dicts):
      merged = {}

      # merge in reverse to preference order
      for d in reversed(dicts):
          merged.update(d)

      return merged

  # RegisterLookupMixin
      def get_lookups(self):
          class_lookups = [getattr(cls, 'class_lookups', {}) for cls in inspect.getmro(self.__class__)]
          return merge_dicts(class_lookups)

      def _get_lookup(self, lookup_name):
          return self.get_lookups().get(lookup_name, None)

# ForeignObject
      def get_lookups(self):
          bases = inspect.getmro(self.__class__)
          bases = bases[:bases.index(ForeignObject)+1]
          class_lookups = [getattr(cls, 'class_lookups', {}) for cls in bases]
  ```